src/mul/generic.jl

```diff
@@ -2,8 +2,12 @@
 # generic methods
 function subspace_mul_generic!(S::AbstractVector{T}, indices, U::AbstractMatrix, subspace, offset=0) where {T}
     D = StrideArrays.static_length(indices)
-    y = StrideArray{T}(undef, (D, ))
+    y = U isa Diagonal ? nothing : StrideArray{T}(undef, (D, ))
     if D == 4
+        # TODO:
+        # this "if" branch is pre-empted by "if size(U, 1) == 4" in subspace_mul!
+        # making subspace_mul_kernel_generic_4x4! redundant
+        # Should we keep subspace_mul4x4! or subspace_mul_kernel_generic_4x4! ?
         for k in subspace
             subspace_mul_kernel_generic_4x4!(S, y, indices, U, k, offset)
         end
```

Comment:
The benchmark results for the `Diagonal`-specialized method of `subspace_mul_kernel_generic_4x4!` are not shown, because this method is always pre-empted by `subspace_mul4x4!`. Should we keep `subspace_mul4x4!` or `subspace_mul_kernel_generic_4x4!` ?

Reply:
We could work on this later. Now we have to focus on the codegen part.

Reply:
We could work on this later. Now we have to focus on the codegen part.